### DISCUSSION TITLE
npm run build is throwing error

### DISCUSSION THREAD

**Discussion by vbp-2610** [fingerprintjs/fingerprintjs#726] [General]
Hi Guys,

I have following version in my mac book 

node ->v16.8.0
npm ->7.21.0
tsc->4.5.0 (type script)

when I am running npm run build, I am getting following issue 

@fingerprintjs/fingerprintjs@3.3.2 build

> rimraf dist && rollup -c

./src/index.ts → dist/fp.js, dist/fp.min.js, dist/fp.umd.js, dist/fp.umd.min.js...

(!) Plugin typescript: @rollup/plugin-typescript TS2345: Argument of type 'unknown' is not assignable to parameter of type 'Readonly<Error>'.

playground/index.ts: (55:37)

55     const errorData = errorToObject(error)

                                       ~~~~~



(!) Plugin typescript: @rollup/plugin-typescript TS2322: Type '(value: unknown) => void' is not assignable to type '() => void'.

resources/content_blocking/make_selectors_tester.ts: (37:50)



37   const abortPromise = new Promise((resolve) => (abort = resolve))

                                                    ~~~~~

(!) Plugin typescript: @rollup/plugin-typescript TS2571: Object is of type 'unknown'.

resources/content_blocking/make_selectors_tester.ts: (45:89)

45           throw new Error(`Failed to fetch filter "${filter.title}" (${filter.file}): ${error.message}`)

                                                                                           ~~~~~



resources/content_blocking/selectors_tester.ts: (52:34)



52     display.value = `${error}\n${error.stack}`

                                    ~~~~~

resources/content_blocking/utils.ts: (85:76)

85         throw new Error(`Failed to fetch a sub-filter (${subFilterURL}): ${error.message}`)

                                                                              ~~~~~



(!) Plugin typescript: @rollup/plugin-typescript TS2339: Property 'zoom' does not exist on type 'CSSStyleDeclaration'.

src/sources/font_preferences.ts: (151:24)



151       iframeBody.style.zoom = `${1 / iframeWindow.devicePixelRatio}`

                           ~~~~



src/sources/font_preferences.ts: (153:24)



153       iframeBody.style.zoom = 'reset'

                           ~~~~



(!) Plugin typescript: @rollup/plugin-typescript TS2551: Property 'msMaxTouchPoints' does not exist on type 'Navigator'. Did you mean 'maxTouchPoints'?

src/sources/touch_support.ts: (25:16)



25   } else if (n.msMaxTouchPoints !== undefined) {

                  ~~~~~~~~~~~~~~~~

  node_modules/typescript/lib/lib.dom.d.ts:9550:14

    9550     readonly maxTouchPoints: number;

                      ~~~~~~~~~~~~~~

    'maxTouchPoints' is declared here.



src/sources/touch_support.ts: (26:24)



26     maxTouchPoints = n.msMaxTouchPoints

                          ~~~~~~~~~~~~~~~~

  node_modules/typescript/lib/lib.dom.d.ts:9550:14

    9550     readonly maxTouchPoints: number;

                      ~~~~~~~~~~~~~~

    'maxTouchPoints' is declared here.



(!) Plugin typescript: @rollup/plugin-typescript TS2322: Type 'unknown' is not assignable to type 'boolean'.

src/utils/async.ts: (20:3)



20   return value && typeof (value as PromiseLike<T>).then === 'function'

     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

(!) Plugin typescript: @rollup/plugin-typescript TS2794: Expected 1 arguments, but got 0. Did you forget to include 'void' in your type argument to 'Promise'?

src/utils/dom.ts: (33:9)

33         _resolve()

           ~~~~~~~~~~



  node_modules/typescript/lib/lib.es2015.promise.d.ts:33:34

    33     new <T>(executor: (resolve: (value: T | PromiseLike<T>) => void, reject: (reason?: any) => void) => void): Promise<T>;

                                        ~~~~~~~~~~~~~~~~~~~~~~~~~

    An argument for 'value' was not provided.

created dist/fp.js, dist/fp.min.js, dist/fp.umd.js, dist/fp.umd.min.js in 3.3s

./src/index.ts → dist/fp.cjs.js, dist/fp.esm.js...

created dist/fp.cjs.js, dist/fp.esm.js in 247ms

./src/index.ts → dist/fp.d.ts...

src/sources/touch_support.ts(25,16): error TS2551: Property 'msMaxTouchPoints' does not exist on type 'Navigator'. Did you mean 'maxTouchPoints'?

src/sources/touch_support.ts(26,24): error TS2551: Property 'msMaxTouchPoints' does not exist on type 'Navigator'. Did you mean 'maxTouchPoints'?

src/sources/font_preferences.ts(151,24): error TS2339: Property 'zoom' does not exist on type 'CSSStyleDeclaration'.

src/sources/font_preferences.ts(153,24): error TS2339: Property 'zoom' does not exist on type 'CSSStyleDeclaration'.

src/utils/async.ts(20,3): error TS2322: Type 'unknown' is not assignable to type 'boolean'.

[!] (plugin dts) Error: Could not load /Library/WebServer/Documents/fingerprint2/src/sources/touch_support.ts (imported by src/sources/index.ts): Failed to compile. Check the logs above.

Error: Could not load /Library/WebServer/Documents/fingerprint2/src/sources/touch_support.ts (imported by src/sources/index.ts): Failed to compile. Check the logs above.

    at error (/Library/WebServer/Documents/fingerprint2/node_modules/rollup/dist/shared/rollup.js:158:30)

    at throwPluginError (/Library/WebServer/Documents/fingerprint2/node_modules/rollup/dist/shared/rollup.js:21831:12)

    at Object.error (/Library/WebServer/Documents/fingerprint2/node_modules/rollup/dist/shared/rollup.js:22536:20)

    at Object.load (/Library/WebServer/Documents/fingerprint2/node_modules/rollup-plugin-dts/dist/rollup-plugin-dts.js:1397:26)

    at /Library/WebServer/Documents/fingerprint2/node_modules/rollup/dist/shared/rollup.js:22780:37

please help to fix this issue ...

**Comment by Finesse** [comment]
Hi @vbp-2610. As you can see in [Actions pipelines](https://github.com/fingerprintjs/fingerprintjs/runs/4709936533#step:6:5), everything runs smoothly if you follow [the contributing guide](https://github.com/fingerprintjs/fingerprintjs/blob/master/contributing.md).

I suppose either of:

- You use NPM instead of Yarn to install the dependencies
- You use a version of TypeScript installed on your machine globally instead of the version specified by the [yarn.lock](https://github.com/fingerprintjs/fingerprintjs/blob/master/yarn.lock) file of this repository

Please try running `yarn install` in this repository before running `yarn build`. If you don't have Yarn, [install it](https://classic.yarnpkg.com/en/docs/install).
